### PR TITLE
Improve test coverage  bio pdb dice

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #820

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/PDB/Dice.py [hunk 1]
```diff
@@ -5,6 +5,21 @@
...
+the file in PDB format
+
+Example
+--------
+Dice 2BEG.pdb, extract chain "B" from residues 18 to 20.
+>>> from Bio.PDB import Dice
+>>> from Bio.PDB import PDBParser
+>>> parser = PDBParser()
+>>> structure = parser.get_structure("scr", "PDB/2BEG.pdb")
+>>> Dice.extract(structure, "B", 18, 20, "/tmp/2BEG_diced.pdb")
```
Comment (peterjc) [inline Bio/PDB/Dice.py]: Using a hard coded output file does not seem ideal - but generating any files in doctests is not ideal. Perhaps just use ``Tests/test_PDB_Dice.py`` only?

**Comment by peterjc** [comment]
It looks like you've accidentally committed changes to ``Tests/GenBank/qualifier_escaping_write.gb`` somehow?

**Comment by peterjc** [comment]
Doctests making files are problematic (when and how is the file going to be removed), but also your example using ``/tmp/`` breaks on Windows:

```
File "C:\projects\biopython\build\lib.win32-2.7\Bio\PDB\Dice.py", line 18, in Bio.PDB.Dice
Failed example:
    Dice.extract(structure, "B", 18, 20, "/tmp/2BEG_diced.pdb")
Exception raised:
    Traceback (most recent call last):
      File "C:\Py\lib\doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest Bio.PDB.Dice[4]>", line 1, in <module>
        Dice.extract(structure, "B", 18, 20, "/tmp/2BEG_diced.pdb")
      File "C:\projects\biopython\build\lib.win32-2.7\Bio\PDB\Dice.py", line 89, in extract
        io.save(filename, sel)
      File "C:\projects\biopython\build\lib.win32-2.7\Bio\PDB\PDBIO.py", line 183, in save
        fp = open(file, "w")
    IOError: [Errno 2] No such file or directory: '/tmp/2BEG_diced.pdb'
----------------------------------------------------------------------
Ran 472 tests in 1074.267 seconds
```

I would remove this doctest and just use the unittest only.

**Comment by svalqui** [comment]
I have removed the doctest, Yes that change to qualifier_escaping_write.gb had leaked somehow, I have overwritten it from upstream/master.

**Comment by peterjc** [comment]
Looks goods, thank you.

Before merging, I'm just trying to sort out an unrelated problem with PIL / pillow on the master branch which is breaking the AppVeyor tests...

**Comment by peterjc** [comment]
Merged, thanks!

**Pull request merged**